code/modules/projectiles/gun_attachables.dm

```diff
@@ -934,7 +934,7 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	scatter_mod = -10
 	movement_acc_penalty_mod = 0.1
 
-/obj/item/attachable/stock/irremoveable/rifle
+/obj/item/attachable/stock/rifle/irremoveable
 	name = "\improper M412 solid stock"
 	pixel_shift_x = 32
 	pixel_shift_y = 13
```

Comment:
instead of switching the typepaths just add what icon it shall use, irremoveable stocks should start using irremoveable typepath so it's acknowledgeable, it also has no stats from the irremoveable stock which is intended, this was changed by another PR and dind't check for sprite change
```suggestion
	pixel_shift_y = 13
	icon_state = "riflestock"
	attach_icon = "riflestock_a"
```

Reply:
Changed the type path, and added the rifle stock stats to /irremoveable/rifle.